scripts/Synapse/updateECRDataStore.ipynb

```diff
@@ -283,7 +295,30 @@
     "        \"org.apache.spark.sql.delta.catalog.DeltaCatalog\",\n",
     "    )\n",
     "    .getOrCreate()\n",
-    ")\n",
+    ")  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.storage.blob import BlobServiceClient\n",
+    "from datetime import datetime\n",
+    "from azure.identity import ManagedIdentityCredential\n",
+    "from azure.core.credentials import AccessToken\n",
+    "from azure.storage.filedatalake import DataLakeServiceClient\n",
+    "import time\n",
+    "\n",
+    "# Class to allow us to use the spark pool's managed identity to access blobs \n",
+    "# and containers (to move them around and rename them)\n",
+    "class spoof_token:\n",
+    "    def get_token(*args, **kwargs):\n",
+    "        return AccessToken(\n",
+    "            token=mssparkutils.credentials.getToken(audience=\"storage\"),\n",
+    "            expires_on=int(time.time())+60*10 # some random time in future... synapse doesn't document how to get the actual time\n",
```

Comment:
Have you run this in the pipeline? I believe this code works when it's run in a notebook but not in a pipeline. 

@DanPaseltiner 's fix to the service principal might help us get around the issue

Reply:
Ugh no I was just running it in the notebook 😭 . What's the issue with it in a pipeline?

Reply:
Nice catch @m-goggins. I agree we should switch here to using our service principle. @bamader the short answer is that we have run into issues with getting the notebooks to run using a managed identity when they are run via the pipeline. To get around this we can use a service principle and get a credential object using its `client_id` and `client_secret`. 

[This PR](https://github.com/CDCgov/phdi/pull/752/files) add in functionality to the SDK to use our own credential manager class to get a credential object by providing a client ID and client secret. In the notebook we can read these values out of a key vault using the linked service that @m-goggins created. One of the notebooks currently deployed in `dev2` should have an example of reading from the key vault and using the service principle creds.